service/history/transferQueueProcessorBase.go

```diff
@@ -163,9 +165,26 @@ func (t *transferQueueProcessorBase) recordWorkflowStarted(
 		return nil
 	}
 
+	var memo []byte
+	encoding := t.shard.GetEncoding(domainEntry)
+	memoBlob, err := t.serializer.SerializeVisibilityMemo(visibilityMemo, encoding)
+	if err != nil {
+		t.logger.WithFields(bark.Fields{
+			logging.TagErr:                 err.Error(),
+			logging.TagDomainID:            domainID,
+			logging.TagWorkflowExecutionID: execution.GetWorkflowId(),
+			logging.TagWorkflowRunID:       execution.GetRunId(),
+		}).Error("error serialize visibility memo")
+	}
+	if memoBlob != nil {
+		memo = memoBlob.Data
+		encoding = memoBlob.GetEncoding()
+	}
+
 	// publish to kafka
 	if t.visibilityProducer != nil {
-		msg := getVisibilityMessageForOpenExecution(domainID, execution, workflowTypeName, startTimeUnixNano, executionTimeUnixNano, taskID)
+		msg := getVisibilityMessageForOpenExecution(domainID, execution, workflowTypeName, startTimeUnixNano,
```

Comment:
For all Kafka messages we just use thrift structures.  Let's stick to the same convention.